code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -444,15 +444,16 @@
 
 /mob/living/simple_animal/hostile/proc/DestroyObjectsInDirection(direction)
 	var/turf/T = get_step(targets_from, direction)
-	if(T && T.Adjacent(targets_from))
+	if(T?.Adjacent(targets_from))
 		if(CanSmashTurfs(T))
 			T.attack_animal(src)
-		for(var/obj/O in T)
-			if(O.density && environment_smash >= ENVIRONMENT_SMASH_STRUCTURES && !O.IsObscured())
+			return
+	for(var/obj/O in T.contents)
+		if(O?.Adjacent(targets_from))
```

Comment:
don't think the ? is necessary considering this is nested under a for

Reply:
Maybe put the ? Between T.Contents right ?